firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/concurrency/CrashlyticsWorker.java

```diff
@@ -152,13 +147,66 @@ public <T, R> Task<R> submitTask(
   }
 
   /**
-   * Blocks until all current pending tasks have completed, up to 30 seconds. Useful for testing.
+   * Submits a <code>Callable Task</code> followed by a <code>SuccessContinuation</code> for
+   * asynchronous execution on the executor.
+   *
+   * <p>This is useful for submitting a task that must be immediately followed by another task, only
+   * if it was successful, but regardless of more tasks being submitted in parallel.
+   *
+   * <p>Returns a <code>Task</code> which will be resolved upon successful completion of the Task
+   * returned by the callable and continued by the continuation, throws an <code>ExecutionException
+   * </code> if either task throws an exception, or throws a <code>CancellationException</code> if
+   * the task is cancelled.
+   */
+  @CanIgnoreReturnValue
+  public <T, R> Task<R> submitTaskOnSuccess(
```

Comment:
Could we have a unit test case for this? I am wondering how to pass the `successContinuation`?

Reply:
Added!